src/GeneratedPlaylists/GeneratedPlaylist.tsx

```diff
@@ -1,17 +1,45 @@
-import React from 'react'
-
+import React, { Component } from 'react'
+import { CleanedAlbumTrack } from '../utils'
 import { getPlaylist } from '../apiCalls'
 
-const getSongsByGenre = () => {
-  // getPlaylist('hip+hop').then(data => console.log(data));
-}
+interface IProps {}
+interface IState { songs: CleanedAlbumTrack[],  selectedGenre: string }
+
+class GeneratedPlaylist extends Component<IProps, IState> {
+  constructor(props: any) {
+    super(props)
+    this.state = {
+      songs: [],
+      selectedGenre: ""
+    }
+  }
+
+  componentDidMount = () => {
+    getPlaylist(this.state.selectedGenre)
+    .then(data => this.setState({ songs: data }));
+  }
+
+  displaySongs = () => {
+    return this.state.songs.reduce((finalSongs: JSX.Element[], currentSong, index: number): JSX.Element[] => {
+      if (index < 10) {
+        const songToDisplay = 
+          <section key={index}>
+            <h3>{currentSong.songName}</h3>
+          </section>;
+        finalSongs.push(songToDisplay)
```

Comment:
This is great to be able to limit how many songs the user will see, based on the randomizing and cleaning of the album tracks that is happening before. Makes it much more contained for a user.